cmd/bench.go

```diff
@@ -0,0 +1,300 @@
+// Copyright 2020 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+
+	"github.com/open-policy-agent/opa/internal/presentation"
+
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/util"
+)
+
+// benchmarkCommandParams are a superset of evalCommandParams
+// but not all eval options are exposed with flags. Only the
+// ones compatible with running a benchmark.
+type benchmarkCommandParams struct {
+	evalCommandParams
```

Comment:
I always regret using struct embedding. I'm curious about the tradeoff compared to duplicating things here. As is, the comment above is misleading because the implementation around benchmarkCommandParams does not support a superset of the eval flags.

Reply:
I guess the assumption is that if the options are not exposed they are not supported. I'm happy to update the message to clarify that point more.

Reply:
Sure, either way. I think it's just moving the debt around if we put it into another struct. I just don't like inheritance 🤣 Since it's contained to a the cmd package and not exported, it's fine.